internal/pkg/agent/application/actions/handlers/handler_action_diagnostics.go

```diff
@@ -205,6 +222,20 @@ func (h *Diagnostics) runHooks(ctx context.Context) ([]client.DiagnosticFileResu
 		elapsed := time.Since(startTime)
 		h.log.Debugw(fmt.Sprintf("Hook %s execution complete, took %s", hook.Name, elapsed.String()), "hook", hook.Name, "filename", hook.Filename, "elapsed", elapsed.String())
 	}
+	if collectCPU {
+		p, err := diagnostics.CreateCPUProfile(ctx, 30*time.Second)
+		if err != nil {
+			return diags, fmt.Errorf("unable to gather CPU profile: %w", err)
+		}
+		diags = append(diags, client.DiagnosticFileResult{
+			Name:        "cpuprofile",
+			Filename:    "cpu.pprof",
```

Comment:
I copied metadata from the control server: https://github.com/elastic/elastic-agent/blob/main/pkg/control/v2/server/server.go#L214-L221
We use `cpu.pprof`, but all other traces use `*.pprof.gz`.

Should we alter `diagnostics.CreateCPUProfile` in order to compress the resulting trace?

Reply:
It is probably correct to change the name to whatever comes back from a GET to http://localhost:6060/debug/pprof/profile?seconds=30

The parameters here could also be pulled out into a shared set of constants much like the profile duration.